src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md

```diff
@@ -63,25 +66,30 @@ current account.
 ```
 PS C:\>Get-AzureRmSubscription
 
-Subscription Name : Contoso Subscription 1
-SubscriptionId    : yyyy-yyyy-yyyy-yyyy
-TenantId          : xxxx-xxxx-xxxx-xxxx
+Name     : Contoso Subscription 1
+Id       : yyyy-yyyy-yyyy-yyyy
+TenantId : xxxx-xxxx-xxxx-xxxx
+State    : Enabled
 
-Subscription Name : Contoso Subscription 2
-SubscriptionId    : yyyy-yyyy-yyyy-yyyy
-TenantId          : xxxx-xxxx-xxxx-xxxx
+Name     : Contoso Subscription 2
+Id       : yyyy-yyyy-yyyy-yyyy
+TenantId : xxxx-xxxx-xxxx-xxxx
+State    : Enabled
 ```
 
 This command gets all subscriptions in the current tenant that are
 authorized for the current user.
 
 ### Example 4: Change the current context to use a specific subscription
 ```
-PS C:\>Get-AzureRmSubscription -SubscriptionId "xxxx-xxxx-xxxx-xxxx" -TenantId "yyyy-yyyy-yyyy-yyyy" | Set-AzureRmContext
+PS C:\>Get-AzureRmSubscription -SubscriptionId "xxxx-xxxx-xxxx-xxxx" -TenantId "yyyy-yyyy-yyyy-yyyy" | Select @{n='SubscriptionId';e={$_.Id}} | Set-AzureRmContext
```

Comment:
@SteGriff the `Set-AzureRmContext` cmdlet accepts a `PSAzureSubscription` object from the pipeline for the `SubscriptionObject` parameter. Any reason you want to change it to pass the `SubscriptionId` to `Set-AzureRmContext` instead? (this is just out of curiosity; I actually didn't know about the `Select @{n=<name>;e=<expression>}` trick ☺️ )

Reply:
My experience of the command as written is that it doesn't work. Here's what happens for me, maybe you can verify?
~~~~
# Outputs my context unchanged. The subscription is not switched, no error:
Get-AzureRmSubscription -SubscriptionId xxxx -TenantId yyyy | Set-AzureRmContext

# Capturing the object specifically and passing it as context throws error:
$ctx = Get-AzureRmSubscription -SubscriptionId xxxx -TenantId yyyy
Set-AzureRmContext -Context $ctx

Set-AzureRmContext : Cannot bind parameter 'Context'. Cannot convert the "8d6daea2-6038-4360-89a9-9944afa3ff6d" 
value of type "Microsoft.Azure.Commands.Profile.Models.PSAzureSubscription" to type 
"Microsoft.Azure.Commands.Profile.Models.PSAzureContext".
At line:1 char:29
+ Set-AzureRmContext -Context $ctx
+                             ~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-AzureRmContext], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.Azure.Commands.Profile.SetAzureRMContextCo 
   mmand
~~~~

Maybe I'm being stupid but this pipe has never worked for me. I have also tried `Select-AzureRmSubscription` at the end and it behaves the same. That's why I've gotten the habit of extracting the `SubscriptionId` specifically.

Perhaps we're straying into the territory of another issue and I should just put the sample the way you have said?

Reply:
Ah, so `Get-AzureRmSubscription` doesn't return a `PSAzureContext`, it returns a `PSAzureSubscription`, so you'll need to run the following:

Reply:
Also, this works:

```
PS C:\Users\markcowl> Get-AzureRmSubscription | Set-AzureRmContext


Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : node
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud

Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : Azure SDK CI
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud

Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : Node CLI Test
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud
```

Reply:
PowerShell is smart enough to accept pipeline input for a subscription or a context for this cmdlet